lib/app/migrate.ts

```diff
@@ -436,19 +436,21 @@ async function finalizeMigration(
     });
 
     if (isMigrationStatus(error) && error.status === MIGRATION_STATUS.FAILURE) {
-      const errorMessage = error.componentErrors
-        ? `${error.projectErrorDetail}: \n\t- ${error.componentErrors
-            .map(componentError => {
-              const {
-                componentType,
-                errorMessage,
-                developerSymbol: uid,
-              } = componentError;
-
-              return `${componentType}${uid ? ` (${uid})` : ''}: ${errorMessage}`;
-            })
-            .join('\n\t- ')}`
-        : error.projectErrorDetail;
+      const { componentErrors, projectErrorDetail } = error;
+      if (!componentErrors || !componentErrors.length) {
+        throw new Error(projectErrorDetail);
+      }
```

Comment:
I refactored a little while I was here, but the gist of it was we were missing a length check.